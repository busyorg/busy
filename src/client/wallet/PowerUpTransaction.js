import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FormattedMessage, FormattedRelative, FormattedDate, FormattedTime } from 'react-intl';
import BTooltip from '../components/BTooltip';

const PowerUpTransaction = ({ timestamp, amount, to, from }) => (
  <div className="UserWalletTransactions__transaction">
    <div className="UserWalletTransactions__icon-container">
      <i className="iconfont icon-flashlight_fill UserWalletTransactions__icon" />
    </div>
    <div className="UserWalletTransactions__content">
      <div className="UserWalletTransactions__content-recipient">
        <div>
          {to === from ? (
            <FormattedMessage id="powered_up" defaultMessage="Powered up " />
          ) : (
            <FormattedMessage
              id="powered_up_to"
              defaultMessage="Powered up {to} "
              values={{
                to: (
                  <Link to={`/@${to}`}>
                    <span className="username">{to}</span>
                  </Link>
                ),
              }}
            />
          )}
        </div>

        <span className="UserWalletTransactions__payout">{amount}</span>
      </div>
      <span className="UserWalletTransactions__timestamp">
        <BTooltip
          title={
            <span>
              <FormattedDate value={`${timestamp}Z`} /> <FormattedTime value={`${timestamp}Z`} />
            </span>
          }
        >
          <span>
            <FormattedRelative value={`${timestamp}Z`} />
          </span>
        </BTooltip>
      </span>
    </div>
  </div>
);

PowerUpTransaction.propTypes = {
  timestamp: PropTypes.string.isRequired,
  amount: PropTypes.element.isRequired,
  to: PropTypes.string.isRequired,
  from: PropTypes.string.isRequired,
};

export default PowerUpTransaction;
