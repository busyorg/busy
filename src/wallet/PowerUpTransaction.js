import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedRelative } from 'react-intl';

const PowerUpTransaction = ({ timestamp, amount }) => (
  <div className="UserWalletTransactions__transaction">
    <div className="UserWalletTransactions__icon-container">
      <i className="iconfont icon-flashlight UserWalletTransactions__icon" />
    </div>
    <div className="UserWalletTransactions__content">
      <div className="UserWalletTransactions__content-recipient">
        <FormattedMessage id="powered_up" defaultMessage="Powered up " />
        <span className="UserWalletTransactions__payout">
          {amount}
        </span>
      </div>
      <span className="UserWalletTransactions__timestamp">
        <FormattedRelative value={`${timestamp}Z`} />
      </span>
    </div>
  </div>
);

PowerUpTransaction.propTypes = {
  timestamp: PropTypes.string.isRequired,
  amount: PropTypes.element.isRequired,
};

export default PowerUpTransaction;
