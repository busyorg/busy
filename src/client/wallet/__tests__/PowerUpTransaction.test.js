import React from 'react';
import { shallow } from 'enzyme';
import PowerUpTransaction from '../PowerUpTransaction';

describe('(Component) PowerUpTransaction', () => {
  describe('with default prop values', () => {
    it('renders and matches snapshot', () => {
      const props = {
        timestamp: '0',
        amount: <span>{'0 STEEM'}</span>,
        to: 'hellosteem',
        from: 'hellosteem',
      };
      const wrapper = shallow(<PowerUpTransaction {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
