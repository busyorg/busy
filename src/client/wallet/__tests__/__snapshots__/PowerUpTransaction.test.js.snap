// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) PowerUpTransaction with default prop values renders and matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PowerUpTransaction
    amount={
      <span>
        0 STEEM
      </span>
    }
    from="hellosteem"
    timestamp="0"
    to="hellosteem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="UserWalletTransactions__icon-container"
        >
          <i
            className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
          />
        </div>,
        <div
          className="UserWalletTransactions__content"
        >
          <div
            className="UserWalletTransactions__content-recipient"
          >
            <div>
              <FormattedMessage
                defaultMessage="Powered up "
                id="powered_up"
                values={Object {}}
              />
            </div>
            <span
              className="UserWalletTransactions__payout"
            >
              <span>
                0 STEEM
              </span>
            </span>
          </div>
          <span
            className="UserWalletTransactions__timestamp"
          >
            <BTooltip
              title={
                <span>
                  <FormattedDate
                    value="0Z"
                  />
                   
                  <FormattedTime
                    value="0Z"
                  />
                </span>
              }
            >
              <span>
                <FormattedRelative
                  updateInterval={10000}
                  value="0Z"
                />
              </span>
            </BTooltip>
          </span>
        </div>,
      ],
      "className": "UserWalletTransactions__transaction",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
          />,
          "className": "UserWalletTransactions__icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-flashlight_fill UserWalletTransactions__icon",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <div>
                <FormattedMessage
                  defaultMessage="Powered up "
                  id="powered_up"
                  values={Object {}}
                />
              </div>
              <span
                className="UserWalletTransactions__payout"
              >
                <span>
                  0 STEEM
                </span>
              </span>
            </div>,
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>,
          ],
          "className": "UserWalletTransactions__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div>
                  <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />
                </div>,
                <span
                  className="UserWalletTransactions__payout"
                >
                  <span>
                    0 STEEM
                  </span>
                </span>,
              ],
              "className": "UserWalletTransactions__content-recipient",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "defaultMessage": "Powered up ",
                    "id": "powered_up",
                    "values": Object {},
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                    0 STEEM
                  </span>,
                  "className": "UserWalletTransactions__payout",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0 STEEM",
                  },
                  "ref": null,
                  "rendered": "0 STEEM",
                  "type": "span",
                },
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>,
              "className": "UserWalletTransactions__timestamp",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>,
                "title": <span>
                  <FormattedDate
                    value="0Z"
                  />
                   
                  <FormattedTime
                    value="0Z"
                  />
                </span>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "updateInterval": 10000,
                    "value": "0Z",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              "type": [Function],
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="UserWalletTransactions__icon-container"
          >
            <i
              className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
            />
          </div>,
          <div
            className="UserWalletTransactions__content"
          >
            <div
              className="UserWalletTransactions__content-recipient"
            >
              <div>
                <FormattedMessage
                  defaultMessage="Powered up "
                  id="powered_up"
                  values={Object {}}
                />
              </div>
              <span
                className="UserWalletTransactions__payout"
              >
                <span>
                  0 STEEM
                </span>
              </span>
            </div>
            <span
              className="UserWalletTransactions__timestamp"
            >
              <BTooltip
                title={
                  <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>
                }
              >
                <span>
                  <FormattedRelative
                    updateInterval={10000}
                    value="0Z"
                  />
                </span>
              </BTooltip>
            </span>
          </div>,
        ],
        "className": "UserWalletTransactions__transaction",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="iconfont icon-flashlight_fill UserWalletTransactions__icon"
            />,
            "className": "UserWalletTransactions__icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "iconfont icon-flashlight_fill UserWalletTransactions__icon",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="UserWalletTransactions__content-recipient"
              >
                <div>
                  <FormattedMessage
                    defaultMessage="Powered up "
                    id="powered_up"
                    values={Object {}}
                  />
                </div>
                <span
                  className="UserWalletTransactions__payout"
                >
                  <span>
                    0 STEEM
                  </span>
                </span>
              </div>,
              <span
                className="UserWalletTransactions__timestamp"
              >
                <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />
                       
                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>
              </span>,
            ],
            "className": "UserWalletTransactions__content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <FormattedMessage
                      defaultMessage="Powered up "
                      id="powered_up"
                      values={Object {}}
                    />
                  </div>,
                  <span
                    className="UserWalletTransactions__payout"
                  >
                    <span>
                      0 STEEM
                    </span>
                  </span>,
                ],
                "className": "UserWalletTransactions__content-recipient",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedMessage
                      defaultMessage="Powered up "
                      id="powered_up"
                      values={Object {}}
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "defaultMessage": "Powered up ",
                      "id": "powered_up",
                      "values": Object {},
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                      0 STEEM
                    </span>,
                    "className": "UserWalletTransactions__payout",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "0 STEEM",
                    },
                    "ref": null,
                    "rendered": "0 STEEM",
                    "type": "span",
                  },
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <BTooltip
                  title={
                    <span>
                      <FormattedDate
                        value="0Z"
                      />
                       
                      <FormattedTime
                        value="0Z"
                      />
                    </span>
                  }
                >
                  <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>
                </BTooltip>,
                "className": "UserWalletTransactions__timestamp",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />
                  </span>,
                  "title": <span>
                    <FormattedDate
                      value="0Z"
                    />
                     
                    <FormattedTime
                      value="0Z"
                    />
                  </span>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <FormattedRelative
                      updateInterval={10000}
                      value="0Z"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "updateInterval": 10000,
                      "value": "0Z",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
